src/compas_slicer/post_processing/generate_raft.py

```diff
@@ -0,0 +1,178 @@
+import logging
+import math
+
+import compas_slicer
+from compas_slicer.geometry import Layer
+from compas_slicer.geometry import Path
+
+from compas.geometry import Point
+from compas.geometry import Line
+from compas.geometry import Vector
+from compas.geometry import bounding_box_xy
+from compas.geometry import offset_polygon
+from compas.geometry import intersection_line_line
+from compas.geometry import offset_line
+
+logger = logging.getLogger('logger')
+
+__all__ = ['generate_raft']
+
+
+def generate_raft(slicer,
+                  raft_offset=10,
+                  distance_between_paths=10,
+                  direction="horizontal",
+                  raft_layers=1,
+                  raft_layer_height="default"):
+    """Creates a raft.
+
+    Parameters
+    ----------
+    slicer: :class:`compas_slicer.slicers.BaseSlicer`
+        An instance of one of the compas_slicer.slicers.BaseSlicer classes.
+    raft_offset: float
+        Distance (in mm) that the raft should be offsetted from the first layer. Defaults to 10mm
+    distance_between_paths: float
+        Distance (in mm) between the printed lines of the raft. Defaults to 10mm
+    direction: str
+        horizontal: Create a raft in the horizontal direction
+        vertical: Create a raft in the vertical direction
+        diagonal: Create a raft int the diagonal direction
+    raft_layers: int
+        Number of raft layers to add. Defaults to 1
+    raft_layer_height: float
```

Comment:
Is this a float or a string?
If it is a float, then we shouldn't set the default value to a string.
If we need a value that corresponds to default, then maybe we could use `raft_layer_height=-1` so we stay with numbers?

Reply:
Ah yes I see how this is confusing. But I wanted to set the default value to the same layer height that is used for slicing, since that makes the most sense if nothing is specified. Therefore these lines:

```
if raft_layer_height == "default":
        raft_layer_height = slicer.layer_height
```

I tried to just call slicer.layer_height in the arguments but this does not work. If you know a better way to achieve this?

Reply:
Yes makes sense. I just have my linter complaining that the parameters should be a float, but it is a string. Another option would be to set the default value to `raft_layer_height = None`. Then you could change the if statement to:

```
if not raft_layer_height:
        raft_layer_height = slicer.layer_height
```

Reply:
ah yes I could change it to that.